### ISSUE TITLE
Implement categorical cross entropy loss

### ISSUE THREAD

**Issue by wmeints** [wmeints/Neuromatic#24]

**Comment by wmeints** [comment]
Blocked on a problem in TensorFlow. The gradient operator for softmax with cross entropy is not implemented in C++. Going to attempt to fix it and compile the right binaries.
